doc/source/rllib/core-concepts.rst

```diff
@@ -163,176 +163,162 @@ These batches are wrapped up together in a ``MultiAgentBatch``,
-                min_batch_size=config["train_batch_size"])) \
-            .for_each(TrainOneStep(workers))
-
-        # type: LocalIterator[ResultDict]
-        return StandardMetricsReporting(train_op, workers, config)
-
+    def training_iteration(self) -> ResultDict:
+        # type: SampleBatchType
+        train_batch = synchronous_parallel_sample(
+                        worker_set=self.workers,
+                        max_env_steps=self.config["train_batch_size"]
+                    )
 
-As you can see, each step returns an *iterator* over objects (if you're unfamiliar with distributed iterators, see Ray's `parallel iterators implementation <https://github.com/ray-project/ray/blob/master/python/ray/util/iter.py>`__).
-The reason it is a ``LocalIterator`` is that, though it is based on a parallel computation, the iterator has been turned into one that can be consumed locally in sequence by the program.
-A couple other points to note:
+        # type: ResultDict
+        train_results = train_one_step(self, train_batch)
 
- - The reason the plan returns an iterator over training results, is that ``trainer.train()`` is pulling results from this iterator to return as the result of the train call.
- - The rollout workers have been already created ahead of time in the ``WorkerSet``, so the execution plan function is only defining a sequence of operations over the results of the rollouts.
+        # Update worker weights with the weights from the trained
+        # policy on the local worker.
+        self.workers.sync_weights()
 
-These iterators represent the infinite stream of data items that can be produced from the dataflow.
-Each operator (e.g., ``ConcatBatches``, ``TrainOneStep``), executes an operation over each item and returns a transformed item (e.g., concatenated batches, learner stats from training). Finally, some operators such as TrainOneStep have the *side-effect* of updating the rollout worker weights (that's why ``TrainOneStep`` takes the list of worker actors ``workers`` as an argument).
-
-Understanding and Debugging Execution Plans
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-Execution plans are based on Ray `parallel iterators <https://github.com/ray-project/ray/blob/master/python/ray/util/iter.py>`__ and can be inspected similarly. For example, suppose you wanted to print out the intermediate data items during training. This can be done by inserting a print function into the dataflow, e.g., for A2C:
-
-.. code-block:: python
+        return train_results
 
-    def debug_print(item):
-        print("I saw", type(item))
-        return item
+As you can see each call to ``training_iteration`` does one sampling and training update. A dictionary is a returned that contains the results of the training update.
+ Its generally recommended that the dictionary map keys of type ``str`` to values that are of type ``float`` or to dictionaries of the same form, allowing for a nested structure.
```

Comment:
I think this is one of the most cryptic things still around in RLlib: The returned metrics dicts. How do we explain the users what these should look like? :)
Maybe for a start, can we give a simple example for a metrics dict here in prettyprint format?
Just use the one returned by PG's training iteration and slim it down a little (e.g. throw out all the empty custom dicts in there).

Reply:
+1 on metrics being cryptic!

Reply:
I put something in there lmk what you think